cm_tools.drush.inc

```diff
@@ -0,0 +1,328 @@
+    if (in_array('field_data_' . $field_name, $all_tables, TRUE)) {
+      foreach ($column_names as $column_name => $replacement) {
+        $addressfield_queries[] = strtr("UPDATE field_data_${field_name} SET %column = $replacement WHERE %column IS NOT NULL AND %column <> ''", array('%column' => $field_name . '_' . $column_name));
+      }
+    }
+    if (in_array('field_revision_' . $field_name, $all_tables, TRUE)) {
+      foreach ($column_names as $column_name => $replacement) {
+        $addressfield_queries[] = strtr("UPDATE field_revision_${field_name} SET %column = $replacement WHERE %column IS NOT NULL AND %column <> ''", array('%column' => $field_name . '_' . $column_name));
+      }
+    }
+  }
+
+  if (!empty($addressfield_queries)) {
+    drush_sql_register_post_sync_op('addressfield', dt('Sanitize Addressfield.'), implode(';', $addressfield_queries) . ';');
+  }
+}
+
+/**
+ * Get field names of a specific field type.
+ *
+ * @param string $field_type
+ *   The field type to search for.
+ *
+ * @return string[]
+ *   An array of field names.
+ */
+function _cm_tools_drush_get_fields_of_type($field_type) {
+  return _cm_tools_drush_db_query("SELECT field_name FROM field_config WHERE type = ':field_type'", array(':field_type' => $field_type));
+}
+
+/**
+ * Get roles which have a specific permission.
+ *
+ * @param string $permission
+ *   The permission to fetch roles for.
+ *
+ * @return int[]
+ *   An array of role IDs with the given permission
+ */
+function _cm_tools_drush_get_roles_with_permission($permission) {
+  return _cm_tools_drush_db_query("SELECT rid FROM role_permission WHERE permission = ':permission'", array(':permission' => $permission));
+}
+
+/**
+ * Get users with specified roles.
+ *
+ * @param int[] $roles
+ *   The roles to return users for.
+ *
+ * @return int[]
+ *   An array of user IDs with the specified roles.
+ */
+function _cm_tools_drush_get_users_with_roles(array $roles) {
+  return _cm_tools_drush_db_query('SELECT DISTINCT uid FROM users_roles WHERE rid IN (:roles)', array(':roles' => implode(',', $roles)));
+}
+
+/**
+ * Execute a SQL query in a limited Drush bootstrap environment.
+ *
+ * @param string $query
+ *   The SQL query to execute.
+ * @param array $placeholders
+ *   An array of placeholders to replace in the query.
+ *
+ * @return array
+ *   An array of lines returned from executing the SQL query.
+ */
+function _cm_tools_drush_db_query($query, array $placeholders) {
+  $prepared_query = strtr(rtrim($query, ' ;') . ';', $placeholders);
+  if (drush_sql_get_class()->query($prepared_query)) {
+    return drush_shell_exec_output();
+  }
+  else {
+    return array();
+  }
+}
+
+/**
+ * Get users with specified permissions.
+ *
+ * @param string $permission
+ *   The permission to return users for.
+ *
+ * @return int[]
+ *   An array of users with the specified permissions.
+ */
+function _cm_tools_drush_get_users_with_permission($permission) {
```

Comment:
This function might be useful for sites, where there is user data stored in other fields and whatnot, maybe we should make this an API function and have it lying around somewhere central for use.

Reply:
👍